state/upgrades_test.go

```diff
@@ -331,19 +331,16 @@ func (s *upgradesSuite) TestStripLocalUserDomainLastConnection(c *gc.C) {
 }
 
 func (s *upgradesSuite) assertStrippedUserData(c *gc.C, coll *mgo.Collection, expected []bson.M) {
-	s.assertUpgradedData(c, StripLocalUserDomain, expectUpgradedData{coll, expected})
+	s.assertUpgradedData(c, StripLocalUserDomain, expectUpgradedData{coll, expected, nil})
 }
 
 type expectUpgradedData struct {
 	coll     *mgo.Collection
 	expected []bson.M
+	filter   bson.D
 }
 
 func (s *upgradesSuite) assertUpgradedData(c *gc.C, upgrade func(*StatePool) error, expect ...expectUpgradedData) {
-	s.assertUpgradedDataWithFilter(c, upgrade, nil, expect...)
-}
-
-func (s *upgradesSuite) assertUpgradedDataWithFilter(c *gc.C, upgrade func(*StatePool) error, filter bson.D, expect ...expectUpgradedData) {
```

Comment:
why remove this? it's a nice helper and the changes require "nil" to be used everywhere and reading the code at the various call points it's no longer "obvious" what's being done

Reply:
It had filtering at the wrong level - the filter is dependent on the specific collection the data comes from, but this has one filter and applies it to all of the collections. 
I can re-add it if you want but it seems like having a filter per expectUpgradedData is a more sensible way to do it.